src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -228,13 +214,13 @@ def interpreter_constraints(self) -> tuple[str, ...]:
             """
         ),
     )
-    pip_version = EnumOption(
-        default=PipVersion.V23_1_2,
+    pip_version = StrOption(
+        default="23.1.2",
         help=softwrap(
             """
             Use this version of Pip for resolving requirements and generating lockfiles.
 
-            N.B.: The `latest` value selects the latest of the listed choices which is not
+            N.B.: The `latest` value selects the latest of the choices listed by PEX which is not
```

Comment:
I think now that there is no enum we should explain that the string you use here must be one of the Pip versions supported by the underlying PEX version. And we should point to: 1) how to find that PEX version 2) how to find the list of Pip versions that PEX version supports and 3) how to upgrade PEX.

Probably the best way to do this is to add an "Setting Pex and Pip versions" section to https://www.pantsbuild.org/docs/pex-files and linking to it here (via the doc_url() macro, which you can see lots of examples of all over the codebase).

Reply:
Suggested edit: https://www.pantsbuild.org/v2.17/suggested-edits/64e4f6906d44d2004b372651
- Note: I retitled https://www.pantsbuild.org/v2.17/docs/pex-files from "Pex files" to "Pex" to reflect that the page isn't only about Pex files anymore. Perhaps that should come with a URL change as well to https://www.pantsbuild.org/v2.17/docs/pex.
- Note: I made the suggested edit to v2.17 documentation specifically.

Reply:
Thanks, but we now generate docs from sources in the repo, so you can make the edit in this PR!

Reply:
Oh wow! I didn't know that! Thanks for pointing that out!

I rolled the suggested edit into this PR. I tried to mimic the syntax I saw used elsewhere (looks like it's some kind of custom Markdown?), so it'd be good to get a second pair of eyes on the syntax and formatting.